async-nats/src/lib.rs

```diff
@@ -887,6 +892,8 @@ impl Subscriber {
             uid,
             sender,
             receiver,
+            max: None,
+            delivered: 0,
```

Comment:
If we want to track this locally, that would have to be on the `Subscription` side so that reconnect can send unsubscribe with an adjusted count. Out of scope for this PR imo.

```suggestion
```

Reply:
Slight correction, we need to track and close when the condition hits but on the sender side in subscription.

Reply:
Agreed on approach on slack. Will update PR.